posthog/api/app_metrics.py

```diff
@@ -27,28 +32,24 @@ class AppMetricsViewSet(TeamAndOrgViewSetMixin, mixins.RetrieveModelMixin, views
 
     def retrieve(self, request: request.Request, *args: Any, **kwargs: Any) -> response.Response:
         try:
-            # probe if we have a valid uuid, and thus are requesting metrics for a batch export
-            uuid.UUID(kwargs["pk"])
+            rows = self.get_batch_export_runs_app_metrics_queryset(batch_export_id=kwargs["pk"])
+
+            dates = [row["dates"].strftime("%Y-%m-%d") for row in rows]
+            successes = [row["successes"] for row in rows]
+            failures = [row["failures"] for row in rows]
```

Comment:
QuerySets hit the database only "the first time we iterate over it". However, there are some minimal performance gains from doing this in a single loop, list comprehensions just were more natural for me to write. I can refactor.